### ISSUE TITLE
Turbolinks causing accessibility problems

### ISSUE THREAD

**Issue by justcolin** [projecthydra/sufia#1585]
While doing some accessibility tests on U-M's upcoming Sufia 7 website I noticed that Turbolinks is used for Sufia. Turbolinks causes a number of accessibility problems, especially for screen reader users who are not notified that a new page is being loaded, when the new page is loaded, and about what new content is available. It also has the more basic problem that **all** users are not notified that the page is being refreshed, which causes confusion. I can go into more detail about specific problems if you all are interested.

- The easy solution is to remove Turbolinks. Turbolinks shouldn't have a major performance benefit for an application like Sufia its removal would automatically make a bunch of accessibility problems go away.
- The hard solution would be to add custom page loading bars, focus management, and a lot of WAI-ARIA tags which will only be supported by some browsers/assistive-technologies. In my experience that is a very brittle solution, which is less accessible than the easy solution, and a lot harder to maintain/test.

We are going with the easy solution but this seemed like the kind of thing that the Sufia community and maybe the UI Working Group should know about and have a discussion about.

Cheers!

**Comment by mjgiarlo** [comment]
@justcolin We do recommend the easy solution in our README: https://github.com/projecthydra/sufia#user-interface Is there anything else we should do in the meantime, short of the hard and brittle solution you mention?

**Comment by justcolin** [comment]
@mjgiarlo Sorry for missing that! The instance I am working with was setup by someone else so I never went though that document.

The section you linked to has a typo: the file that references Turbolinks is in `app/assets/stylesheets/application.js` not `app/assets/stylesheets/application.css`. You may also want to add a line saying something like, "We recommend disabling Turbolinks due to a number of accessibility problems it creates."

Is that line in `application.js` being added by Sufia, Blacklight or the Turbolinks Gem? Is it Sufia or Blacklight that is requiring the Turbolinks Gem (or both)? (Apologies if these sound naive.)

**Comment by justcolin** [comment]
I should note that I would never recommend the hard solution—there be many dragons there. Just having Turbolinks removed from Sufia's dependancies and codebase would be great and should be pretty easy.

Having it installed but disabled by default is a possibility, but I don't like the practice of giving people a tempting way to make their websites inaccessible.

**Comment by mjgiarlo** [comment]
@justcolin Done: https://github.com/projecthydra/sufia/commit/d5e496b6cce62ac130b6442cd02050b7efff05d5

I believe it's Rails that is including turbolinks by default.

Thanks!

**Comment by justcolin** [comment]
@mjgiarlo: Thanks! Now I just have to convince the whole Rails community that Turbolinks is a bad idea. That shouldn't be too hard. :-P
